### PR TITLE
when offbording don't remove webhook from gitlab if other component is

### PR DESCRIPTION
**Pull request by rcerven**
using same repository as well

[KONFLUX-5944](https://issues.redhat.com//browse/KONFLUX-5944)

### Checklist:
 - PR has reference to the issue(s) it resolves
 - Write / update unit tests
 - Write / update integration (envtest) tests
 - Ensure there is an issue for e2e tests if needed
 - Ensure `make test` passes
 - Ensure test coverage hasn't decreased
 - Test code changes manually
 - Update readme and documentation
 - Write PR description that highlights overall changes and add usage examples if applicable

### CODE REVIEW

File: controllers/component_build_controller_pac.go [hunk 1]
```diff
@@ -535,7 +536,26 @@ func (r *ComponentBuildReconciler) UnconfigureRepositoryForPaC(ctx context.Conte
...
+				continue
+			}
+			componentUrl := strings.TrimSuffix(strings.TrimSuffix(comp.Spec.Source.GitSource.URL, ".git"), "/")
+			if componentUrl == gitUrl {
+				sameRepoUsed = true
+				break
+			}
+		}
+
+		if webhookTargetUrl != "" && !sameRepoUsed {
```
Comment (mmorhun) [inline controllers/component_build_controller_pac.go:558]: It doesn't make sense to determine `sameRepoUsed` if `webhookTargetUrl` is empty string.

File: controllers/component_build_controller_pac.go [hunk 2]
```diff
@@ -535,7 +536,26 @@ func (r *ComponentBuildReconciler) UnconfigureRepositoryForPaC(ctx context.Conte
 
 	isAppUsed := IsPaCApplicationConfigured(gitProvider, pacConfig)
 	if !isAppUsed {
-		if webhookTargetUrl != "" {
+		componentList := &appstudiov1alpha1.ComponentList{}
+		if err := r.Client.List(ctx, componentList, &client.ListOptions{Namespace: component.Namespace}); err != nil {
+			log.Error(err, "failed to list components")
+			return "", "", "", err
+		}
+		gitUrl := strings.TrimSuffix(strings.TrimSuffix(component.Spec.Source.GitSource.URL, ".git"), "/")
```
Comment (mmorhun) [inline controllers/component_build_controller_pac.go:544]: nitpick: this line sorta separates operation on components list. I'd move it at the first line of the block (before working with the component list).

File: controllers/component_build_controller_pac.go (hunk 1 above)
Comment (rcerven) [reply controllers/component_build_controller_pac.go:558]: we delete webhook ONLY if
webhookurl isn't empty (so we have some webhook)
AND we don't have other component with the same URL

Comment (rcerven) [reply controllers/component_build_controller_pac.go]: fixed

**Review by mmorhun — APPROVED** [review]

**Pull request merged**